bqplot/traits.py

```diff
@@ -112,18 +114,20 @@ def _to_json(a, obj=None):
             if np.issubdtype(a.dtype, np.float):
                 # replace nan with None
                 dtype = a.dtype
-                a = np.where(np.isnan(a), None, a)
+                a = np.where(np.isnan(a), 0, a)
             elif a.dtype in (int, np.int64):
                 dtype = 'float'
                 a = a.astype(np.float64)
             elif np.issubdtype(a.dtype, np.datetime64):
                 dtype = 'date'
                 a = a.astype(np.str)
+            elif a.size == 0:
+                a = np.empty(0)
             else:
                 dtype = a.dtype
-            return {'values': a.tolist(), 'type': str(dtype)}
+            return {'values': a.tolist(), 'data': b64encode(a), 'type': str(a.dtype), 'shape': a.shape, 'size': a.size}
```

Comment:
Is `values` supposed to remain included here? Otherwise this seems to be *more* work, rather than less.